src/python/pants/util/contextutil.py

```diff
@@ -22,6 +22,13 @@
 from pants.util.tarutil import TarFile
 
 
+class InvalidZipPath(zipfile.BadZipfile):
+  """Raise instead of passing unchecked falsey values through to zipfile.ZipFile.
```

Comment:
I was hoping to motivate it's existence  alongside the upstream BadZipfile - without that info the distinction is tough to describe.

But that was all an attempt to not break the public API behavior of raising BadZipfile. I will update as requested, since I have been persuaded by your argument that since it wasn't documented, the exception type can be fair game :)